query-algebrizer/src/clauses/resolve.rs

```diff
@@ -73,11 +82,16 @@ impl ConjoiningClauses {
         use self::FnArg::*;
         match arg {
             FnArg::Variable(var) => {
-                self.constrain_var_to_type(var.clone(), ValueType::Instant);
-                self.column_bindings
-                    .get(&var)
-                    .and_then(|cols| cols.first().map(|col| QueryValue::Column(col.clone())))
-                    .ok_or_else(|| Error::from_kind(ErrorKind::UnboundVariable(var.name())))
+                match self.bound_value(&var) {
+                    Some(TypedValue::Instant(v)) => Ok(QueryValue::TypedValue(TypedValue::Instant(v))),
+                    _ => {
```

Comment:
Same review comment as before:

> This isn't quite right. If it's already bound to the wrong kind of value, we'll still fall through to the column binding and get an `UnboundVariable` error.

You need a similar `bail` line here as there is on line 53.